src/main/java/teammates/common/datatransfer/questions/FeedbackRubricQuestionDetails.java

```diff
@@ -619,43 +620,24 @@ public String getQuestionResultStatisticsHtml(List<FeedbackResponseAttributes> r
-        DecimalFormat df = new DecimalFormat("#");
-        DecimalFormat dfAverage = new DecimalFormat("0.00");
+        StringBuilder tableBodyHtml = getQuestionResultsStatisticsBodyHtml(fqd, responseFrequency, rubricStats);
 
-        for (int i = 0; i < numOfRubricSubQuestions; i++) {
-            StringBuilder tableBodyFragmentHtml = new StringBuilder();
-            boolean isSubQuestionRespondedTo = responseFrequency[i][numOfRubricChoices] > 0;
-
-            for (int j = 0; j < numOfRubricChoices; j++) {
-                String percentageFrequencyString = isSubQuestionRespondedTo
-                                                 ? df.format(rubricStats[i][j] * 100) + "%"
-                                                 : STATISTICS_NO_VALUE_STRING;
-                String tableBodyCell = Templates.populateTemplate(tableBodyFragmentTemplate,
-                        Slots.RUBRIC_PERCENTAGE_FREQUENCY_OR_AVERAGE,
-                        percentageFrequencyString + " (" + responseFrequency[i][j] + ")");
-                tableBodyFragmentHtml.append(tableBodyCell).append(System.lineSeparator());
-            }
+        StringBuilder tableBodyExcludingSelfHtml;
 
-            if (fqd.hasAssignedWeights) {
-                String averageString = isSubQuestionRespondedTo
-                                     ? dfAverage.format(rubricStats[i][numOfRubricChoices])
-                                     : STATISTICS_NO_VALUE_STRING;
-                String tableBodyAverageCell = Templates.populateTemplate(tableBodyFragmentTemplate,
-                        Slots.RUBRIC_PERCENTAGE_FREQUENCY_OR_AVERAGE, averageString);
-                tableBodyFragmentHtml.append(tableBodyAverageCell).append(System.lineSeparator());
-            }
+        if (isExcludingSelfOptionAvailable) {
 
-            // Get entire row
-            String tableRow = Templates.populateTemplate(tableBodyTemplate,
-                    Slots.SUB_QUESTION, StringHelper.integerToLowerCaseAlphabeticalIndex(i + 1) + ") "
-                            + SanitizationHelper.sanitizeForHtml(rubricSubQuestions.get(i)),
-                    Slots.RUBRIC_ROW_BODY_FRAGMENTS, tableBodyFragmentHtml.toString());
-            tableBodyHtml.append(tableRow).append(System.lineSeparator());
+            int[][] responseFrequencyExcludingSelf =
+                    RubricStatistics.calculateResponseFrequencyExcludingSelf(responsesForStatistics, fqd);
+            float[][] rubricStatsExcludingSelf = RubricStatistics.calculatePercentageFrequencyAndAverage(fqd,
+                    responseFrequencyExcludingSelf);
+            tableBodyExcludingSelfHtml = getQuestionResultsStatisticsBodyHtml(fqd,
+                    responseFrequencyExcludingSelf, rubricStatsExcludingSelf);
+        } else {
+            tableBodyExcludingSelfHtml = new StringBuilder();
+            tableBodyExcludingSelfHtml.append("").append(System.lineSeparator());
```

Comment:
Why do we need `.append("").append(System.lineSeparator());`?

Reply:
I just wanted to explicitly show that when the checkbox isn't available, the div is filled with empty line. 

Actually this should suffice `tableBodyExcludingSelfHtml.append(System.lineSeparator());` ? Or do you want me to get rid of the whole thing?